### PR TITLE
dstor: Removed unnecessary pack/unpack operation

### PR DESCRIPTION
**Pull request by karasevb**
Restored the part of PR #260

Thanks to @hjelmn for the contribution!

Signed-off-by: Boris Karasev <karasev.b@gmail.com>

### CODE REVIEW

**Comment by karasevb** [comment]
@artpol84 @rhc54 please review

**Comment by jjhursey** [comment]
@karasevb Please PR to `v1.2` branch. Thanks!

**Comment by artpol84** [comment]
@karasevb this looks ok.
Please test this PR additionally with OMPI as @rhc54 suggested offlist and merge once tested.

File: src/dstore/pmix_esh.c [hunk 1]
```diff
@@ -2245,30 +2242,21 @@ static int _store_data_for_rank(ns_track_elem_t *ns_info, pmix_rank_t rank, pmix
...
-            }
-            PMIX_RELEASE(kp); // maintain acctg - hash_store does a retain
-            cnt = 1;
-            kp = PMIX_NEW(pmix_kval_t);
-        }
-        cnt = 1;
-        PMIX_RELEASE(kp);
-        PMIX_RELEASE(bptr);  // free's the data region
-        if (PMIX_ERR_UNPACK_READ_PAST_END_OF_BUFFER != rc) {
+    bptr = buf;
```
Comment (artpol84) [inline src/dstore/pmix_esh.c]: @karasevb as a minor cleanup - it seems that we don't need `bptr` anymore and `buf` itself should be fine.

**Pull request merged**